src/chess/chess_game.rs

```diff
@@ -518,6 +519,96 @@ impl IGame for ChessGame {
     fn get_repetition_limit() -> Option<u32> {
         Some(3)
     }
+
+    fn produce_transformed_data_entries(entry: DataEntry<Self>) -> Vec<DataEntry<Self>> {
+        let transform =
+            |e: &DataEntry<Self>, transform_sq: &dyn Fn(chess::Square) -> chess::Square| {
+                let b = &e.pos.board;
+                let pieces = b
+                    .combined()
+                    .into_iter()
+                    .map(|square| {
+                        (
+                            transform_sq(square),
+                            b.piece_on(square).unwrap(),
+                            b.color_on(square).unwrap(),
+                        )
+                    })
+                    .collect_vec();
+
+                let board = chess::Board::try_from(chess::BoardBuilder::setup(
+                    pieces.iter(),
+                    b.side_to_move(),
+                    b.castle_rights(chess::Color::White),
+                    b.castle_rights(chess::Color::Black),
+                    b.en_passant().map(|square| transform_sq(square).get_file()),
+                ))
+                .expect("unable to transform board");
+                let pos = ChessPosition {
+                    board,
+                    fifty_rule_count: e.pos.fifty_rule_count,
+                };
+
+                let probs = e
+                    .probs
+                    .iter()
+                    .map(|(m, p)| {
+                        (
+                            ChessMove::new(chess::ChessMove::new(
+                                transform_sq(m.m.get_source()),
+                                transform_sq(m.m.get_dest()),
+                                m.m.get_promotion(),
+                            )),
+                            *p,
+                        )
+                    })
+                    .collect_vec();
+
+                let winner = e.winner;
+                DataEntry { pos, probs, winner }
+            };
+
+        let rows_mirror = |e: &DataEntry<Self>| {
+            transform(e, &|sq| {
+                chess::Square::make_square(
+                    chess::Rank::from_index(7 - sq.get_rank().to_index()),
+                    sq.get_file(),
+                )
+            })
+        };
+        let columns_mirror = |e: &DataEntry<Self>| {

```

Comment:
are we okay with creating a board with a weird "setup"? (queen on dark columns)
if so let's add a comment about it because it's unintuitive

Reply:
what do you mean weird? what are dark columns? all the transposed position can be reached by legal moves from the regular starting position

Reply:
haha sorry I was in a hurry when I wrote it. 
I meant if you take the regular starting position and you do a columns_mirror, the queens are not on their color (color of the משבצת) and the kings are (which is a "weird" position)
not the the color of the tile matters so much but still, can't be reached by legal moves from the regular starting position

Reply:
ahhh now i see you only do it when there are no castle rights

Reply:
1. i think this is a counterexample (i.e. a reflection that cannot happen in a real game)
2. i think we don't care
do you agree?

<img width="305" alt="image" src="https://user-images.githubusercontent.com/4618146/200163820-1d7ca6a9-7efa-48b1-a473-4498cba9f3c1.png">

Reply:
i don't think we care about these, the point it all the transformed positions are completely legal, and the (val, probs) is relevant for them

Reply:
sababot